x/mongo/driver/topology/topology_test.go

```diff
@@ -230,6 +235,30 @@ func TestServerSelection(t *testing.T) {
 			t.Fatalf("did not receive error from server selection")
 		}
 	})
+	t.Run("Context Deadline Error", func(t *testing.T) {
+		desc := description.Topology{}
+		topo, err := New()
+		noerr(t, err)
+		subCh := make(chan description.Topology, 1)
+		subCh <- desc
+		resp := make(chan error)
+		timeout := make(chan time.Time)
+		go func() {
```

Comment:
I assume you copied this pattern of running a goroutine and selecting over a results channel and a `time.After` from the test above. This is fine, but I think you can use `assert.Soon` here to clean up the code a bit: https://github.com/mongodb/mongo-go-driver/blob/a6cd2226c7836c1264a00b7101f751e95eeef570/internal/testutil/assert/assert.go#L86.

Reply:
done